### PR TITLE
feat: update dashboard insights to use API v2 with histograms

### LINKED ISSUE #2508: Feature: v2 API - use `v2/histogram/prs` in place of `v1/prs/insights`
### Type of feature

🍕 Feature

### Current behavior

Currently, `v1/prs/insights` grabs blocks of data in for `accepted_prs`, `closed_prs`, etc. based on a few time buckets (typically 30 days wide). This is _very_ close to what we can do with `v2/histogram/pull-requests` and a wider time bucket.

### Suggested solution

For example, hitting the `v2/historgram/prs` endpoint for the `open-sauced/app` repo:

```sh
curl -X 'GET' \
  'https://alpha.opensauced.pizza/v2/histogram/pull-requests?range=90&prev_days_start_date=0&width=90&repo=open-sauced%2Fapp' \
  -H 'accept: application/json'
```

Returns data that looks like:
```json
[
  {
    "bucket": "2023-11-28T00:00:00.000Z",
    "prs_count": 109,
    "accepted_prs": 75,
    "closed_prs": 14,
    "draft_prs": 9,
    "active_prs": 9,
    "spam_prs": 0,
    "pr_velocity": 3
  },
  {
    "bucket": "2023-10-14T00:00:00.000Z",
    "prs_count": 151,
    "accepted_prs": 130,
    "closed_prs": 9,
    "draft_prs": 5,
    "active_prs": 5,
    "spam_prs": 0,
    "pr_velocity": 3
  },
  {
    "bucket": "2023-08-30T00:00:00.000Z",
    "prs_count": 2,
    "accepted_prs": 0,
    "closed_prs": 1,
    "draft_prs": 0,
    "active_prs": 0,
    "spam_prs": 0,
    "pr_velocity": 0
  }
]
```

If these bucket times/widths look abit strange, it's because timescale is doing most of the heavy lifting trying to figure out where the "origin" is for the given data and the associated events. Read more on this here: https://docs.timescale.com/use-timescale/latest/time-buckets/about-time-buckets/#origin

---

Related to: 
- https://github.com/open-sauced/api/pull/470
- https://github.com/open-sauced/app/issues/2507

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Migrates the dashboard insights to the new `/v2/histogram/pull-requests` endpoint
- Deletes previous `useInsights` hook
- Needs https://github.com/open-sauced/api/issues/496 to be resolved first

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #2508 

## Mobile & Desktop Screenshots/Recordings

No visual changes, but different data.

image

## Steps to QA

1. Click on https://deploy-preview-2516--oss-insights.netlify.app/pages/brandonroberts/5/dashboard?range=30
2. Note that PR insights are still displayed

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: components/organisms/Dashboard/dashboard.tsx [hunk 1]
```diff
@@ -15,13 +15,13 @@ interface DashboardProps {
...
 const Dashboard = ({ repositories }: DashboardProps): JSX.Element => {
-  const { data: insightsData, isLoading } = useInsights(repositories);
+  const { data: insightsData, isLoading } = usePullRequestsHistogram(repositories);
   const { data: contributorData, meta: contributorMeta } = useContributors(undefined, repositories);
   const router = useRouter();
   const { range = 30 } = router.query;
 
-  const compare1 = getInsights(insightsData, 30);
-  const compare2 = getInsights(insightsData, 60);
+  const compare1 = getInsights(insightsData, 0);
```
Comment (nickytonline) [inline components/organisms/Dashboard/dashboard.tsx:23]: Does the interval argument, `0`, mean get today's insights and `1`, mean yesterdays?

Comment (brandonroberts) [reply components/organisms/Dashboard/dashboard.tsx:23]: Yes, the first item in the array is today, and second is yesterday. The interval used to be returned as part of the data. I'll update the naming

File: lib/hooks/api/usePullRequestsHistogram.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      acceptedTotal: 0,
+      spamPercentage: 0,
+      acceptedPercentage: 0,
+      unlabeledPrsTotal: 0,
+      unlabeledPercentage: 0,
+    };
+  }
+
+  const spam =
+    (currentInsights.prs_count || 0) > 0 ? Math.round((currentInsights.spam_prs / currentInsights.prs_count) * 100) : 0;
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply lib/hooks/api/usePullRequestsHistogram.ts:33]: Its probably not necessary anymore

**Review by nickytonline — APPROVED** [review]
Looks good @brandonroberts! :shipit:

**Review by jpmcb — APPROVED** [review]

**Comment by jpmcb** [comment]
Looks like this closes https://github.com/open-sauced/app/issues/2508 ? 👏🏼

**Pull request merged**